File: backend/app/schemas/credentials.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, SecretStr


class DashScopeCredentialWrite(BaseModel):
    api_key: SecretStr


class ProviderCredentialRead(BaseModel):
    provider: str = "DASHSCOPE"
    configured: bool
    key_hint: str | None = None
    verified: bool = False
    verified_at: datetime | None = None
    updated_at: datetime | None = None


class ProviderCredentialVerificationRead(BaseModel):
    provider: str = "DASHSCOPE"
    status: Literal[
        "VERIFIED",
        "INVALID",
        "FORBIDDEN",
        "RATE_LIMITED",
        "UNAVAILABLE",
    ]
    verified: bool
    key_hint: str
    verified_at: datetime | None = None
    message: str

File: frontend/src/types/credentials.ts
export type ProviderCredential = {
  provider: "DASHSCOPE";
  configured: boolean;
  key_hint: string | null;
  region: "cn-beijing";
  provider_workspace_id: string | null;
  verified: boolean;
  verified_at: string | null;
  updated_at: string | null;
};

export type CredentialVerificationStatus =
  | "VERIFIED"
  | "INVALID"
  | "FORBIDDEN"
  | "RATE_LIMITED"
  | "UNAVAILABLE";

export type CredentialVerification = {
  provider: "DASHSCOPE";
  status: CredentialVerificationStatus;
  verified: boolean;
  key_hint: string;
  region: "cn-beijing";
  provider_workspace_id: string | null;
  verified_at: string | null;
  message: string;
};
